### PR TITLE
Emit channel close event on ordered channel close

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #202

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by AdityaSripal — APPROVED** [review]
Nice work!

File: modules/core/04-channel/keeper/timeout.go [hunk 1]
```diff
@@ -156,6 +156,7 @@ func (k Keeper) TimeoutExecuted(
 	if channel.Ordering == types.ORDERED {
 		channel.State = types.CLOSED
 		k.SetChannel(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), channel)
+		EmitChannelClosedEvent(ctx, packet, channel)
```
Comment (crodriguezvega) [inline modules/core/04-channel/keeper/timeout.go:159]: @colin-axner recommended in his [comment](https://github.com/cosmos/ibc-go/issues/202#issuecomment-1123647973) to do the event emission after `EmitTimeoutPacketEvent(ctx, packet, channel)` below. Would you mind moving it there? I guess you need to wrap it in a check to make sure that the channel is ordered and is closed...

Comment (chatton) [reply modules/core/04-channel/keeper/timeout.go]: Ah I missed that part of the comment, thanks @crodriguezvega, I've made the appropriate changes

Comment (crodriguezvega) [reply modules/core/04-channel/keeper/timeout.go]: Thanks, @chatton!

**Review by crodriguezvega — APPROVED** [review]
🥇

**Comment by crodriguezvega** [comment]
Ay, just realized that we need a changelog entry, @chatton! :)

File: modules/core/04-channel/keeper/timeout.go (hunk 1 above)
Comment (colin-axner) [reply modules/core/04-channel/keeper/timeout.go]: Just as a note, I actually disagree with my past comment. Emitting after closing the channel works just fine imo

**Pull request merged**